File: ruby/testcode/deserial_015.rb
require_relative 'shared'

module Oj
  def self.load(input, opts = {}); input; end
end

# vuln-code-snippet start ruby_deser_oj_object
def parse_oj_object(req)
  data = Oj.load(req.body_str, mode: :object) # vuln-code-snippet vuln-line ruby_deser_oj_object
  BenchmarkResponse.ok(data.to_s)
end
# vuln-code-snippet end ruby_deser_oj_object

File: php/testcode/deserial_015.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_deser_file_upload_content
function deserial015(BenchmarkRequest $req): BenchmarkResponse {
    $file = $req->file('import');
    $content = file_get_contents($file['tmp_name']);
    $data = unserialize($content); // vuln-code-snippet vuln-line php_deser_file_upload_content
    return BenchmarkResponse::json(['imported' => count($data)]);
}
// vuln-code-snippet end php_deser_file_upload_content
